compiler/quilt/vfs.py

```diff
@@ -106,31 +106,73 @@ def open_func(filename, mode='r', module=module, *args, **kwargs):
     patcher.start()
     return patcher
 
+
+# Note: To test this, uninstall h5py, then run your quilt_vfs_test.py.
+def scrub_patchmap(patchmap, verbose=False, strict=False):
+    """Return a version of patchmap with missing modules/callables removed.
+
+    :param verbose: Mention each dropped module and/or function
+    :param strict: Raise an error if module exists but func doesn't
+    """
+    result = {}
+
+    for modname, funcname in patchmap.items():
+        try:
+            module = importlib.import_module(modname)
+        except ImportError:
+            if verbose:
+                print("Dropped {} from the patch map (module not present)".format(modname))
+            continue
+        if not hasattr(module, funcname):
+            if verbose:
+                print("Dropped {}.{} from the patch map (func/class not present)")
+            if strict:
+                raise AttributeError("{} is missing from module {}".format(funcname, modname))
+            continue
+        if not callable(getattr(module, funcname)):
+            if verbose:
+                print("Dropped {}.{} from the patch map (not callable)")
+            if strict:
+                raise TypeError("{}.{} is not callable".format(funcname, modname))
+            continue
+        result[modname] = funcname
+    return result
+
+
 DEFAULT_MODULE_MAPPINGS = {
-    'builtins': 'open', 'bz2': 'BZ2File', 'gzip': 'GzipFile'
-    # for keras/tensorflow
-    ,'h5py': 'File',
+    'builtins': 'open',
+    'bz2': 'BZ2File',
+    'gzip': 'GzipFile',
+    'h5py': 'File',    # for keras/tensorflow
     # manually add tensorflow because it's a heavyweight library
-    #'tensorflow': 'gfile.FastGFile', 'tensorflow': 'gfile.GFile'
+    #'tensorflow': 'gfile.FastGFile', 
+    #'tensorflow': 'gfile.GFile',
 }
+DEFAULT_MODULE_MAPPINGS = scrub_patchmap(DEFAULT_MODULE_MAPPINGS, True)
+
 
 # simple mapping of illegal chars to underscores.
 # TODO: more sophisticated function for handling illegal identifiers, e.g. number as first char
 DEFAULT_CHAR_MAPPINGS = dict([(char, '_') for char in string.whitespace + string.punctuation])
 
+
 def make_mapfunc(pkg, hash=None, version=None, tag=None, force=False,
                  mappings=None, install=False, charmap=DEFAULT_CHAR_MAPPINGS, **kwargs):
     """core support for mapping filepaths to objects in quilt local objs/ datastore.
        TODO: add support for reading/iterating directories, e.g. os.scandir() and friends
     """
-    if len(kwargs) == 0:
-        kwargs = DEFAULT_MODULE_MAPPINGS
+#XXX: Unused
+    # if len(kwargs) == 0:
+    #     kwargs = DEFAULT_MODULE_MAPPINGS
     if install:
         command.install(pkg, hash=hash, version=version, tag=tag, force=force)
     owner, pkg = parse_package(pkg)
+
     if mappings is None:
         mappings = { ".": "" }  # TODO: test this case
-    pkgname = "quilt.data."+owner+"."+pkg
+#XXX: Unused/debug
+#    pkgname = "quilt.data."+owner+"."+pkg
+
```

Comment:
obviously, remove...

Reply:
No prob, didn't know if they were left in there for dev/debug purposes.  ..not my baby, don't want to mess with it too much.  :-)